src/diffusers/schedulers/scheduling_dpmsolver_multistep.py

```diff
@@ -242,10 +242,10 @@ def set_timesteps(self, num_inference_steps: int = None, device: Union[str, torc
                 np.linspace(0, last_timestep - 1, num_inference_steps + 1).round()[::-1][:-1].copy().astype(np.int64)
             )
         elif self.config.timestep_spacing == "leading":
-            step_ratio = last_timestep // (num_inference_steps + 1)
+            step_ratio = last_timestep // num_inference_steps
             # creates integer timesteps by multiplying by ratio
             # casting to int to avoid issues when num_inference_step is power of 3
-            timesteps = (np.arange(0, num_inference_steps + 1) * step_ratio).round()[::-1][:-1].copy().astype(np.int64)
+            timesteps = (np.arange(0, num_inference_steps) * step_ratio).round()[::-1].copy().astype(np.int64)
```

Comment:
Why are we doing these changes? This doesn't look to be backwards compatible

Reply:
We made this change to be consistent with how timesteps are sampled with other schedulers: 
https://github.com/huggingface/diffusers/blob/acd1962769715cebee1bb8e87075db4e78490070/src/diffusers/schedulers/scheduling_ddim.py#L325
https://github.com/huggingface/diffusers/blob/acd1962769715cebee1bb8e87075db4e78490070/src/diffusers/schedulers/scheduling_ddpm.py#L307
Is there any specific reason why this scheduler differs here?

Reply:
The current implementation will mean that the scheduler does not end at timestep "0" (or "1"), which we observed to result in artifacts during generation. 
Which btw. also results in `DPMSolverMultistepScheduler` and `DPMSolverMultistepInverseScheduler` yielding different sets of steps.